apps/ts.c

```diff
@@ -888,7 +889,8 @@ static TS_VERIFY_CTX *create_verify_ctx(const char *data, const char *digest,
         f = TS_VFY_VERSION | TS_VFY_SIGNER;
         if (data != NULL) {
             f |= TS_VFY_DATA;
-            if (TS_VERIFY_CTX_set_data(ctx, BIO_new_file(data, "rb")) == NULL)
+            out = BIO_new_file(data, "rb");
+            if (TS_VERIFY_CTX_set_data(ctx, out) == NULL)
                 goto err;
```

Comment:
No, only this one. If the call to `TS_VERIFY_CTX_set_data()` succeeds then ownership of the BIO has been transferred to the `TS_VERIFY_CTX` and will get freed when `TS_VERIFY_CTX_free()` gets called (which happens in the `err` block in the event of an error).